drivers/display/display_gc9x01x.c

```diff
@@ -0,0 +1,745 @@
+	enum display_orientation orientation;
+
+	/* Set all the required registers. */
+	retval = config->regs_init_fn(dev);
+	if (retval < 0) {
+		return retval;
+	}
+
+	/* pixel format */
+	if (config->pixel_format == GC9X01X_PIXEL_FORMAT_RGB565) {
+		pixel_format = PIXEL_FORMAT_RGB_565;
+	} else {
+		pixel_format = PIXEL_FORMAT_RGB_888;
+	}
+
+	/* Set pixel format. */
+	retval = gc9x01x_set_pixel_format(dev, pixel_format);
+	if (retval < 0) {
+		return retval;
+	}
+
+	/* orientation */
+	if (config->rotation == GC9X01X_ORIENTATION_90) {
+		orientation = DISPLAY_ORIENTATION_ROTATED_90;
+	} else if (config->rotation == GC9X01X_ORIENTATION_180) {
+		orientation = DISPLAY_ORIENTATION_ROTATED_180;
+	} else if (config->rotation == GC9X01X_ORIENTATION_270) {
+		orientation = DISPLAY_ORIENTATION_ROTATED_270;
+	} else {
+		orientation = DISPLAY_ORIENTATION_NORMAL;
+	}
+
+	/* Set display orientation. */
+	retval = gc9x01x_set_orientation(dev, orientation);
+	if (retval < 0) {
+		return retval;
+	}
+
+	if (config->inversion) {
+		/* Display inversion mode. */
+		retval = gc9x01x_transmit(dev, GC9X01X_CMD_INVON, NULL, 0);
+		if (retval < 0) {
+			return retval;
+		}
+	}
+
+	return 0;
+}
+
+static int gc9x01x_set_brightness(const struct device *dev, const uint8_t brightness)
+{
+	const struct gc9x01x_config *config = dev->config;
+
+	if (config->backlight.dev != NULL) {
+		uint32_t step = config->backlight.period / 100U;
+		int retval = pwm_set_pulse_dt(&config->backlight, brightness * step / 2.55);
+
+		if (retval) {
+			LOG_ERR("Failed to set pulse width\n");
+			return retval;
+		}
+		return 0;
+	}
+
+	return 0;
+}
+
+static int gc9x01x_init(const struct device *dev)
+{
+	const struct gc9x01x_config *config = dev->config;
+
+	int retval;
+
+	if (!spi_is_ready_dt(&config->spi)) {
+		LOG_ERR("SPI device is not ready");
+		return -ENODEV;
+	}
+
+	if (!gpio_is_ready_dt(&config->cmd_data)) {
+		LOG_ERR("Command/Data GPIO device not ready");
+		return -ENODEV;
+	}
+
+	retval = gpio_pin_configure_dt(&config->cmd_data, GPIO_OUTPUT);
+	if (retval < 0) {
+		LOG_ERR("Could not configure command/data GPIO (%d)", retval);
+		return retval;
+	}
+
+	if (config->reset.port != NULL) {
+		if (!device_is_ready(config->reset.port)) {
+			LOG_ERR("Reset GPIO device not ready");
+			return -ENODEV;
+		}
+
+		retval = gpio_pin_configure_dt(&config->reset, GPIO_OUTPUT_INACTIVE);
+		if (retval < 0) {
+			LOG_ERR("Could not configure reset GPIO (%d)", retval);
+			return retval;
+		}
+	}
+
+	gc9x01x_hw_reset(dev);
+
+	gc9x01x_display_blanking_on(dev);
+
+	retval = gc9x01x_configure(dev);
+	if (retval < 0) {
+		LOG_ERR("Could not configure display (%d)", retval);
+		return retval;
+	}
+
+	retval = gc9x01x_exit_sleep(dev);
+	if (retval < 0) {
+		LOG_ERR("Could not exit sleep mode (%d)", retval);
+		return retval;
+	}
+
+	retval = gc9x01x_set_brightness(dev, 50);
+	if (retval < 0) {
+		LOG_ERR("Could not set brightness (%d)", retval);
+		return retval;
+	}
```

Comment:
it should be up to the application to enable the backlight at the right moment, see my comment above

Reply:
It may stay a proprietary solution until a pwm backlight init driver is available (See above).
For better clarity I added a separate devicetree priority to define the initial value.